src/adapter/src/coord/timestamp_selection.rs

```diff
@@ -506,7 +506,7 @@ impl Coordinator {
             //
             // This should only happen for literals that have no sources or sources that
             // are known to have completed (non-tailed files for example).
-            Timestamp::MAX
+            i64::MAX.try_into().expect("known to fit")
```

Comment:
I think we should either add a comment or new-type to explain why `i64` is being used.

Reply:
I'm also somewhat worried that there are other places where a timestamp can advance past `i64::MAX`. What about in the `write_ts` method itself?

Reply:
Oh yeah, this is just to try and pinpoint where the problem is. Sorry I wasn't clear about that! 🙈